presto-main/src/main/java/io/prestosql/execution/SqlStageExecution.java

```diff
@@ -524,6 +536,12 @@ else if (taskState == TaskState.FINISHED) {
         }
     }
 
+    private synchronized boolean isFlushing()
+    {
+        return flushingTasks.size() > 0
```

Comment:
Add a comment explaining that to be flushing, there must be at least one flushing task, and all others must be flushing or finished.

also replace `flushingTasks.size() > 0` with `!flushingTasks.isEmpty()`